dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -398,8 +398,8 @@ Path findClassLocation(String className) {
       try {
         JavaClass javaClass = classRepository.loadClass(classFileName);
         javaClasses.add(javaClass);
-      } catch (ClassNotFoundException ex) {
-        // We couldn't find the class in the jar file where we found it.
+      } catch (ClassNotFoundException | ClassFormatException ex) {
```

Comment:
Is a ClassNotFoundException usually caused by a corrupted jar file? Or is that just a normal version conflict where the version of the jar we have is not the one that was expected?

Reply:
ClassNotFoundException was also caused by a corrupted JAR file as in
https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/1090/files

Reply:
Yes, I'm just wondering if there are causes that are not corrupt jar files, and whether we're properly reporting them if so? It's weird that #1090 used ClassNotFoundException instead of ClassFormatException. How certain is it that that that was the right fix?

Reply:
In the kinesis-client problem (#1090), JAR index contained a file starting with ".", which was weird. BCEL threw ClassNotFoundException because it cannot find such file in the JAR. So the fix was to catch ClassNotFoundException. I see the fix was reasonable.

Reply:
Was it really a corrupt jar file or was BCEL buggy here?

I don't think all ClassNotFoundExceptions indicate corrupt jar files, even if that particular one does.

Reply:
As discussed, javap also show "class not found" for the kinesis JAR.

```
suztomo@suxtomo24:~$ javap -cp /usr/local/google/home/suztomo/.m2/repository/com/amazonaws/amazon-kinesis-client/1.13.0/amazon-kinesis-client-1.13.0.jar '.unison.com.e007f77498fd27177e2ea931a06dcf50.unison.tmp/amazonaws/services/kinesis/multilang/messages/CheckpointMessage'
Error: class not found: .unison.com.e007f77498fd27177e2ea931a06dcf50.unison.tmp/amazonaws/services/kinesis/multilang/messages/CheckpointMessage

```